Makefile

```diff
@@ -0,0 +1,31 @@
+SHELL := /bin/sh
+BOLD := \033[1m
+DIM := \033[2m
+RESET := \033[0m
+
+.PHONY: build lint mockgen unit-tests integration-tests test all
+
+build:
+	@echo -e "$(BOLD)Buiding$(RESET)"
+	@go install ./...
+
+lint:
+	@echo -e "$(BOLD)Linting source code$(RESET)"
+	@golangci-lint run
+
+mockgen:
+	@echo -e "$(BOLD)Generating mocking code$(RESET)"
+	@go get github.com/golang/mock/mockgen@v1.4.4
+	@./run_mockgen.sh
```

Comment:
does this really only ever have to be run once? Doesn't it have to be updated if the mocks need to be rebuilt? can we make this an actual (vs. phoney) make target on the actual files that need to be updated vs. their source files?